model/riscv_insts_zbc.sail

```diff
@@ -20,12 +20,8 @@ mapping clause assembly = RISCV_CLMUL(rs2, rs1, rd)
   <-> "clmul" ^ spc() ^ reg_name(rd) ^ sep() ^ reg_name(rs1) ^ sep() ^ reg_name(rs2)
 
 function clause execute (RISCV_CLMUL(rs2, rs1, rd)) = {
-  let rs1_val = X(rs1);
-  let rs2_val = X(rs2);
-  var result : xlenbits = zeros();
-  foreach (i from 0 to (xlen_val - 1))
-    if rs2_val[i] == bitone then result = result ^ (rs1_val << i);
-  X(rd) = result;
+  let prod = carryless_mul(X(rs1), X(rs2));
+  X(rd) = truncate(prod, xlen);
```

Comment:
I think `prod[xlen - 1 .. 0]` is a bit clearer and also more type safe IIRC. Ditto other places.